internal/parser.go

```diff
@@ -2612,6 +2611,14 @@ func parseForeignContent(p *parser) bool {
 			}
 		}
 		return true
+	case StartExpressionToken:
+		p.reconstructActiveFormattingElements()
+		p.addExpression()
```

Comment:
Just FYI we do have `expressionIM`, but I think we can circle back to implementing it here. I tried it out and it was causing some problems with `setOriginalIM()` being called twice.

I think our insertion mode algorithm probably needs to be updated to use a stack for improved spec compliance. Again, outside of the scope of this PR.